### ISSUE TITLE
Upgrade: Option to upgrade and pin a pinned package (remove pin, upgrade, add pin)

### ISSUE THREAD

**Issue by Terrance** [chocolatey/choco#3320]
### Checklist

- [X] I have verified this is the correct repository for opening this issue.
- [X] I have verified no other issues exist related to my request.

### Is Your Feature Request Related To A Problem? Please describe.

The process for upgrading pinned packages requires me first to unpin each one with `choco pin remove`, and then run `choco upgrade --pin` \* with the list of packages -- it would be nice to have this all in a single step, e.g. `choco upgrade --move-pin`.

---

\* The output includes no mention of the packages being re-pinned, so I still end up checking `choco pin` afterwards to make sure that's happened -- it would be useful to get a message similar to that from `choco pin add`.

### Describe The Solution. Why is it needed?

I'm on a relatively slow network connection, so I pin a handful of large packages to be upgraded when needed or when I've got the capacity to max out my connection for a significant amount of time.

It would be less error-prone to not have to work through the list of packages to unpin each time, and having to check afterwards that they're all pinned again.

**Comment by TheCakeIsNaOH** [comment]
This appears to be a duplicate of https://github.com/chocolatey/choco/issues/2050, which I have a PR up for here: https://github.com/chocolatey/choco/pull/2464

> The output includes no mention of the packages being re-pinned, so I still end up checking choco pin afterwards to make sure that's happened -- it would be useful to get a message similar to that from choco pin add.

This is a separate feature request, could you open another issue describing this in more detail?

**Comment by Terrance** [comment]
Apologies, yes, that appears to be the same request -- searching issues for `upgrade` and `pin` brought up a lot of results and I must have skipped over that one.

I see your PR was opened 2 years ago -- is it likely to be merged, or is it waiting for more support from users, or anything else?

> This is a separate feature request, could you open another issue describing this in more detail?

Done: #3321

**Comment by TheCakeIsNaOH** [comment]
> I see your PR was opened 2 years ago -- is it likely to be merged, or is it waiting for more support from users, or anything else?

It's waiting on the Chocolatey team to review and test it. During the Chocolatey V2 development period, there was a soft feature freeze for anything not related to the .Net 4.8 and NuGet assembly uplift, so that is in large part why it was not reviewed before. However, V2 is now out and stable, so the next release hopefully will be a be general feature release.

If you are a licensed customer, I'd suggest contacting support, to see if they can give more details, or if they have any workarounds for your use case, as licensed customers get "Prioritization of Bugs and Features" as per https://chocolatey.org/compare

**Comment by TheCakeIsNaOH** [comment]
Duplicate of #2050
